Shell/Twee.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+
+/**
+ * Implements Twee's goal-directed preprocessing technique for equational reasoning.
+ * To quote "Twee: An Equational Theorem Prover" (Smallbone, 2021):
+ *
+ * ----- 
+ * Twee’s frontend can optionally transform the problem to make the prover more goal-directed.
+ * The transformation is simple, but strange.
+ * For every function term f(...) appearing in the goal, we introduce a fresh constant symbol a and add the axiom f(...) = a
+ * For example, if the goal is f(g(a), b) = h(c) we add the axioms f(g(a), b) = sF0, g(a) = sF1, and h(c) = sF2.
+ * Simplification will rewrite the first axiom to f(sF1, b) = sF0 and the goal to sF0 = sF2.
+ * -----
+ * 
+ * We take an intepretation of this for our purposes, NB:
+ *  - any ground term, in a clause derived from the conjecture, is transformed in this way
```

Comment:
Does this header need updating wrt the new functionality?

Reply:
Damn, it did!